homeassistant/components/somfy/__init__.py

```diff
@@ -95,7 +94,10 @@ async def _update_all_devices():
         previous_devices = data[COORDINATOR].data
         # Sometimes Somfy returns an empty list.
         if not devices and previous_devices:
-            raise UpdateFailed("No devices returned")
+            _LOGGER.warning(
+                "No devices returned. Assuming the previous ones are still valid"
```

Comment:
Should we fail ever the server is flakey for 10x?

Reply:
If we fail, it means the devices will become unavailable and it will break any automation. Despite the empty response, the server accepts command. 
This error occurs really often. 5 times (5 min is common)

Reply:
Will the user still know if the api is down?

If the somfy api goes offline or timees out, will the underlying library raise an exception?

Reply:
The underlying is based on requests, so yes if there is a timeout, or the status is not 200, an exception will be raised.
See https://github.com/tetienne/somfy-open-api/blob/master/pymfy/api/somfy_api.py#L58

Reply:
Looks like https://github.com/tetienne/somfy-open-api/blob/master/pymfy/api/somfy_api.py#L71 will raise if we get back invalid json.  I think that will raise on a timeout as well